internal/engine/eval/rego/eval.go

```diff
@@ -134,14 +138,40 @@ func (e *Evaluator) Eval(
 		return nil, fmt.Errorf("could not prepare Rego: %w", err)
 	}
 
-	rs, err := pq.Eval(ctx, rego.EvalInput(&Input{
+	input := &Input{
 		Profile:      pol,
 		Ingested:     obj,
 		OutputFormat: e.cfg.ViolationFormat,
-	}))
+	}
+
+	if err := enrichInputWithEntityProps(input, entity); err != nil {
+		return nil, fmt.Errorf("error adding properties to input: %w", err)
+	}
+
+	rs, err := pq.Eval(ctx, rego.EvalInput(input))
 	if err != nil {
 		return nil, fmt.Errorf("error evaluating profile. Might be wrong input: %w", err)
 	}
 
 	return e.reseval.parseResult(rs, entity)
 }
+
+func enrichInputWithEntityProps(
+	input *Input,
+	entity protoreflect.ProtoMessage,
```

Comment:
What about using an interface to verify if you can get properties? Newer entities would use the EntityInstance protobuf, which is missing, but would also work with that.

Reply:
I used the same type defined [here](https://github.com/mindersec/minder/blob/main/internal/engine/eval/rego/eval.go#L111), but I can map it to that other type/interface if you point me to an example.

Reply:
There is no such interface, you'd need to define a `propertiesFetcher` interface or something of the sort that accepts `GetProperties`